onnxruntime/python/tools/bert/BertOnnxModel.py

```diff
@@ -0,0 +1,959 @@
+                continue
+            children = input_name_to_nodes[add_after_tanh.output[0]]
+            if len(children) != 1 or children[0].op_type != 'Mul':
+                continue
+            mul_after_tanh = children[0]
+
+            mul_half = self.match_parent(mul_after_tanh, 'Mul', None, output_name_to_node)
+            if mul_half is None:
+                continue
+
+            i = self.find_constant_input(mul_half, 0.5)
+            if i < 0:
+                continue
+
+            root_node = self.get_parent(mul_half, 0 if i == 1 else 1, output_name_to_node)
+            if root_node is None:
+                continue
+
+            mul_before_tanh = self.match_parent(tanh_node, 'Mul', 0, output_name_to_node)
+            if mul_before_tanh is None:
+                continue
+
+            i = self.find_constant_input(mul_before_tanh, 0.7978, delta=0.0001)
+            if i < 0:
+                continue
+
+            add_before_tanh = self.match_parent(mul_before_tanh, 'Add', 0 if i == 1 else 1, output_name_to_node)
+            if add_before_tanh is None:
+                continue
+
+            mul_after_pow = self.match_parent(add_before_tanh, 'Mul', None, output_name_to_node, exclude=[root_node])
+            if mul_after_pow is None:
+                continue
+
+            i = self.find_constant_input(mul_after_pow, 0.0447, delta=0.0001)
+            if i < 0:
+                continue
+
+            pow = self.match_parent(mul_after_pow, 'Pow', 0 if i == 1 else 1, output_name_to_node)
+            if pow is None:
+                continue
+
+            if not self.has_constant_input(pow, 3.0):
+                continue
+
+            if pow.input[0] != root_node.output[0]:
+                continue
+
+            subgraph_nodes = [mul_after_tanh, mul_half, add_after_tanh, tanh_node, mul_before_tanh, add_before_tanh, mul_after_pow, pow]
+            if not self.is_safe_to_fuse_nodes(subgraph_nodes, [mul_after_tanh.output[0]], input_name_to_nodes, output_name_to_node):
+                continue
+
+            nodes_to_remove.extend(subgraph_nodes)
+            gelu_node = onnx.helper.make_node(gelu_op_name,
+                inputs=[root_node.output[0]],
+                outputs=mul_after_tanh.output,
+                name=self.create_node_name(gelu_op_name))
+            gelu_node.domain = "com.microsoft"
+            nodes_to_add.append(gelu_node)
+
+        if len(nodes_to_add) > 0:
+            print("Fused {} count: {}".format('Gelu (FastGelu fits better)' if gelu_op_name == 'Gelu' else 'FastGelu', len(nodes_to_add)))
+
+        self.remove_nodes(nodes_to_remove)
+        self.add_nodes(nodes_to_add)
+
+    def fuse_add_bias_gelu(self):
+        input_name_to_nodes = self.input_name_to_nodes()
+        output_name_to_node = self.output_name_to_node()
+        nodes_to_remove = []
+        nodes_to_add = []
+
+        for node in self.get_nodes_by_op_type('FastGelu'):
```

Comment:
Would you like to add code to fuse "Gelu" and "Add" into "BiasGelu"?

Reply:
ORT can fuse it natively

Reply:
Sounds good. It worth add a comment about that (in readme or here).